sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -471,16 +476,23 @@ public virtual Response<TableItem> CreateIfNotExists(CancellationToken cancellat
             scope.Start();
             try
             {
+                var context = new RequestContext { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow };
+                context.AddClassifier((int)HttpStatusCode.Conflict, false);
                 var response = _tableOperations.Create(
-                    new TableProperties { TableName = Name },
-                    null,
-                    _defaultQueryOptions,
-                    cancellationToken);
-                return Response.FromValue(response.Value as TableItem, response.GetRawResponse());
-            }
-            catch (RequestFailedException ex) when (ex.Status == (int)HttpStatusCode.Conflict && ex.ErrorCode != TableErrorCode.TableBeingDeleted)
-            {
-                return default;
+                    RequestContent.Create(new { TableName = Name }),
```

Comment:
I think this makes a few allocations by creating the anonymous type and then the RequestContent and invokes the ObjectSerializer, which is a perf hit - if perf is a concern here, it might be worth looking into writing this to a desired RequestContent subclass directly.

Reply:
> it might be worth looking into writing this to a desired RequestContent subclass directly.

Is there an existing example of this pattern somewhere I could refer to?

Reply:
realized that there is no reason to use an anonymous type here since we still have models :) 

Since this is typically only ever called once per client lifetime, I'm good with not micro-optimizing further.